docs/tutorials/text_prediction/automm_customization.md

```diff
@@ -0,0 +1,236 @@
+```
+
+### optimization.val_check_interval
+How often within one training epoch to check the validation set. Can specify as float or int.
+- pass a float in the range [0.0, 1.0] to check after a fraction of the training epoch.
+- pass an int to check after a fixed number of training batches.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"optimization.val_check_interval": 0.5})
+# check validation set 4 times during a training epoch
+predictor.fit(hyperparameters={"optimization.val_check_interval": 0.25})
+```
+
+### optimization.top_k
+Based on the validation score, choose top k model checkpoints to do model averaging.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"optimization.top_k": 3})
+# use top 5 checkpoints
+predictor.fit(hyperparameters={"optimization.top_k": 5})
+```
+
+### optimization.top_k_average_method
+Use what strategy to average the top k model checkpoints.
+- `"greedy_soup"`: try to add the checkpoints from best to worst into the averaging pool and stop if the averaged checkpoint performance decreases. See [the paper](https://arxiv.org/pdf/2203.05482.pdf) for details.
+- `"uniform_soup"`: average all the top k checkpoints as the final checkpoint.
+- `"best"`: pick the checkpoint with the best validation performance.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"optimization.top_k_average_method": "greedy_soup"})
+# average all the top k checkpoints
+predictor.fit(hyperparameters={"optimization.top_k_average_method": "uniform_soup"})
+```
+
+### optimization.efficient_finetune
+Finetune only a small portion of parameters instead of one whole pretrained backbone.
+- `"bit_fit"`: bias parameters only.
+- `"norm_fit"`: normalization parameters + bias parameters.
+- `"lora"`: LoRA Adaptors. See [this paper](https://arxiv.org/abs/2106.09685) for details.
+- `"lora_bias"`: LoRA Adaptors + bias parameters.
+- `"lora_norm"`: LoRA Adaptors + normalization parameters + bias parameters.
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"optimization.efficient_finetune": None})
+# finetune only bias parameters
+predictor.fit(hyperparameters={"optimization.efficient_finetune": "bit_fit"})
+```
+
+## Environment
+
+### env.num_gpus
+The number of gpus to use.
+```
+# by default, all available gpus are used by AutoMM
+predictor.fit(hyperparameters={"env.num_gpus": -1})
+# use 1 gpu only
+predictor.fit(hyperparameters={"env.num_gpus": 1})
+```
+
+### env.per_gpu_batch_size
+The batch size for each GPU. 
+```
+# default used by AutoMM
+predictor.fit(hyperparameters={"env.per_gpu_batch_size": 8})
+# use batch size 16 per GPU
+predictor.fit(hyperparameters={"env.per_gpu_batch_size": 16})
+```
+
+### env.batch_size
+The batch size to use in each step of training. If `env.batch_size` is larger than `env.per_gpu_batch_size * env.num_gpus`, we accumulate gradients to reach the effective `env.batch_size` before performing one optimization step.
```

Comment:
If `env.batch_size` is larger than `env.per_gpu_batch_size * env.num_gpus`, does it have to be multiples of `env.per_gpu_batch_size * env.num_gpus`. For example, if `per_gpu_batch_size=2`, `num_gpus=8`, when `batch_size` is 20, I'm curious what will gradient accumulation do?